include/rmm/mr/device/system_memory_resource.hpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/cuda_device.hpp>
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/detail/error.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+namespace rmm::mr {
+
+namespace detail {
+/**
+ * @brief Check if system allocated memory (SAM) is supported on the specified device.
+ *
+ * @param device_id The device to check
+ * @return true if SAM is supported on the device, false otherwise
+ */
+static bool is_system_memory_supported(cuda_device_id device_id)
+{
+  int pageableMemoryAccess;
+  RMM_CUDA_TRY(cudaDeviceGetAttribute(
+    &pageableMemoryAccess, cudaDevAttrPageableMemoryAccess, device_id.value()));
```

Comment:
When was this attribute added? If it's later than our minimum CUDA version then this may not compile? I guess it is passing all CI, so that's a good sign.

Reply:
It was in [11.0](https://docs.nvidia.com/cuda/archive/11.0_GA/cuda-runtime-api/group__CUDART__TYPES.html), while RMM requires 11.4+, so I think we are good here.